resources/testharness.js

```diff
@@ -3702,19 +3726,21 @@ policies and contribution forms [3].
             } else {
                 stack = e.filename + ":" + e.lineno + ":" + e.colno;
             }
-            error_handler(message, stack);
+            error_handler(e.error, message, stack);
         }, false);
 
         addEventListener("unhandledrejection", function(e) {
-            var reason;
-            if (e.reason) {
-                reason  = e.reason.message ? e.reason.message : e.reason;
+            var message;
+            if (e.reason && e.reason.message) {
+                message = "Unhandled rejection: " + e.reason.message;
             } else {
-                reason = e;
+                message = "Unhandled rejection";
+            }
+            var stack;
```

Comment:
I've extracted this part of the change into https://github.com/web-platform-tests/wpt/pull/20123 and hope to land that first.

Reply:
Done and rebased.